generate_changelog.sh

```
@@ -1,31 +1,313 @@
+
+# cache the git full log
+CHANGELOG_BUFFER="${TMPDIR:-/tmp}/.changelog_buffer.txt"
+git log "${GIT_RANGE}" --reverse --pretty=format:"COMMIT_START%n%h%n%B%nCOMMIT_END" >"${CHANGELOG_BUFFER}" ; wait ;
+
+RAW_FLAGS_LIST=$(cat <"${CHANGELOG_BUFFER}" | grep -oE "([\[][A-Z]+[]]){1}" | sort -id | uniq -c | sort -rid | grep -oE "([A-Z]+){1}" | sort -id | uniq | sort -rd ; wait ;)
 
-# cache the git log
-CHANGELOG_BUFFER=".chagelog_buffer.txt"
-cat <(git log "${GIT_RANGE}" --reverse --pretty=format:"COMMIT_START%n%h%n%B%nCOMMIT_END") >"${CHANGELOG_BUFFER}" ; wait ;
+RAW_IMPACTED_ISSUES=$(cat <"${CHANGELOG_BUFFER}" | grep -oE "([#]\d+){1}\b" | sort -id | uniq | xargs -L1 -I{} printf "%s, " "{}" ; wait ;)
+RAW_FLAGS_USED=$(cat <(printf "%s\n" "${RAW_FLAGS_LIST}") | xargs -L1 -I{} printf "%s, " "{}" ; wait ;)
```

Shellcheck (0.10.0)

[style] Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.

(SC2002)